### DISCUSSION TITLE
Installation Problem

### DISCUSSION THREAD

**Discussion by Sharif432** [pypr/pysph#370] [Q&A] [answered]
During the installation "pip installs PySPH", I received an error. 

I'm using python 3.9.13 and MSVC 2022

Output: 

 pysph/base/cell_indexing_nnps.cpp(3280): error C2169: 'log2': intrinsic function, cannot be defined
      pysph/base/cell_indexing_nnps.cpp(4017): warning C4244: '=': conversion from 'Py_ssize_t' to 'int', possible loss of data
      pysph/base/cell_indexing_nnps.cpp(4802): warning C4018: '<': signed/unsigned mismatch
      pysph/base/cell_indexing_nnps.cpp(6378): error C2264: 'log2': error in function definition or declaration; function not called
      pysph/base/cell_indexing_nnps.cpp(6392): error C2264: 'log2': error in function definition or declaration; function not called
      error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.34.31933\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2

**Comment by prabhuramachandran** [comment, marked as answer]
Can you provide more details on this please? Is this from a windows powershell? Or is this from WSL? 
This error is unusual, `log2` is only defined within a suitable guard specific to MSVC. Our tests run every week without error on windows as you can see here it ran 3 days ago and the version of MSVC seems the same as yours: https://github.com/pypr/pysph/actions/runs/4149657878/jobs/7178770542

I am not sure it will help but you can also try to install from the latest version of pysph on github either using: `pip install https://github.com/pypr/pysph/zipball/master` or via the usual `git clone` and other commands mentioned in the documentation.
